packages/react-query/src/errorBoundaryUtils.ts

```diff
@@ -25,11 +25,17 @@ export const ensurePreventErrorBoundaryRetry = <
     TQueryKey
   >,
   errorResetBoundary: QueryErrorResetBoundaryValue,
+  query: Query<TQueryFnData, TError, TQueryData, TQueryKey>,
```

Comment:
just seeing this now: to be on par with `getHasError`, we should make this param required, but potentially undefined, and then do a single check in here. sorry about changing my mind here 😅 


```suggestion
  query: Query<TQueryFnData, TError, TQueryData, TQueryKey> | undefined,
```